app/controllers/YourTaxCodeController.scala

```diff
@@ -49,26 +51,50 @@ trait YourTaxCodeController extends TaiBaseController
             val nino = user.person.nino
 
             for {
-              TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]) <- taxAccountService.taxCodeIncomes(nino, year)
-              scottishTaxRateBands <- taxAccountService.scottishBandRates(nino, year, taxCodeIncomes.map(_.taxCode))
+              taxCodeChange <- taxCodeChangeService.taxCodeChange(nino, year)
+              scottishTaxRateBands <- taxAccountService.scottishBandRates(nino, year, taxCodeChange.current.map(_.taxCode))
             } yield {
-              val taxCodeViewModel = TaxCodeViewModel(taxCodeIncomes, scottishTaxRateBands, year)
+              val taxCodeViewModel = TaxCodeViewModel(taxCodeChange.current, scottishTaxRateBands, year)
               Ok(views.html.taxCodeDetails(taxCodeViewModel))
             }
           }
   }
 
-  def prevTaxCodes(year: TaxYear): Action[AnyContent] = taxCodes(year)
+  def prevTaxCodes(year: TaxYear): Action[AnyContent] = authorisedForTai(personService).async {
+    implicit user =>
+      implicit person =>
+        implicit request =>
+
+          if (taxCodeChangeEnabled) {
+            ServiceCheckLite.personDetailsCheck {
+              val nino = user.person.nino
 
+              for {
+                taxCodeChange <- taxCodeChangeService.taxCodeChange(nino, year)
```

Comment:
This doesnt have to be done now but perhaps the TaxCodeService should be pulling this information with scottish tax codes together so we are not duplicating the logic in here and TaxCodeChangeController. 

TaxCodeChange would have to return a model with both TaxCodeChange and ScottishTaxRateBands or at least a tuple.

This scottishTaxRateBands logic may also require moving to the backend but that definitely isn't for now